stellarphot/settings/custom_widgets.py

```diff
@@ -0,0 +1,391 @@
+        self._item_widget.open_nested = False
+        self._item_widget.open_nested = True
+
+        # Update the current state of the widget
+        self._editing = True
+
+    def _set_disable_state_nested_models(self, top, value):
+        """
+        When a one model contains another and the top-level model widget
+        sets disabled=True that does not actually disable the nested model.
+        This method handles that in a crude way by walking the tree of
+        widgets in the top-level model widget and disabling them all.
+
+        Parameters
+        ----------
+
+        top : `ipyautoui.AutoUi`
+            Top-level widget that may have nested models.
+
+        value : bool
+            State that ``disabled`` should be set to.
+        """
+
+        if hasattr(top, "disabled") or isinstance(top, AutoObject):
+            top.disabled = value
+        try:
+            for child in top.children:
+                self._set_disable_state_nested_models(child, value)
+        except AttributeError:
+            # No children...
+            pass
+
+    def _make_new_widget(self):
+        """
+        Make a new widget for the item type and set up actions for the save button.
+        """
+        match self._item_type_name:
+            case "camera" | Camera.__name__:
+                new_widget = ui_generator(Camera)
+            case "observatory" | Observatory.__name__:
+                new_widget = ui_generator(Observatory)
+            case "passband_map" | PassbandMap.__name__:
+                new_widget = ui_generator(PassbandMap)
+
+        def saver():
+            """
+            Tries to save the new item, and if it fails, shows the confirmation widget.
+            """
+            try:
+                self._saved_settings.add_item(new_widget.model(**new_widget.value))
+            except ValueError:
+                # This will happen if the item already exists
+                self._confirm_edit.show()
+
+        def update_choices_and_select_new():
+            """
+            Update the choices after a new item is saved, update the choices
+            and select the new item.
+            """
+            if not self._editing:
+                value_to_select = new_widget.model(**new_widget.value)
+                self._construct_choices()
+                self._choose_existing.value = value_to_select
+
+        # This is the mechanism for adding callbacks to the save button.
+        new_widget.savebuttonbar.fns_onsave_add_action(saver)
+        new_widget.savebuttonbar.fns_onsave_add_action(update_choices_and_select_new)
+        return new_widget
+
+    def _handle_confirmation(self):
+        """
+        Handle the confirmation of a save operation.
+        """
+
+        # Use a closure here to capture the current state of the widget
+        def save_confirm_handler(change):
+            # value of None means the widget has been reset to not answered
```

Comment:
Do you want to clarify this is that the "confirmation" has not been addressed or the "confirmation widget" has been reset to behave as if not answered?  I was confused by this at first and only realized what you meant by reading the behavior below.  Better yet, maybe place a comment in the function 'docstring' above explaining what the desired behavior is as a whole.